### DISCUSSION TITLE
self contain PyInstaller to run on other computer (and/or from USB)

### DISCUSSION THREAD

**Discussion by drakeredwind01** [pyinstaller/pyinstaller#6215] [Help]
**description.**
I got the PyInstaller to work perfectly with only the ".exe" file in an alt location on my computer, but then I tried to transfer it via remote access to my desktop, and it didn't work.

**Describe the solution/alternatives you'd like**
Can you please add 2 options: 
    add more data to the ".exe" file to run standalone on an alt computer (no python installed) or from a USB. or...
    add a "virtual python" or "virtual anaconda" to be placed next to the .exe (or somewhere else) that the .exe can use to run 
        properly

**Additional context**
This is my first enhancement request, and I hope I did this properly.
I also hope Github notifies me if this is responded to.
If it doesn't, you can reach me at drakeredwind01@gmail.com
or even sooner by commenting on one of my videos
    https://www.youtube.com/channel/UCvPJKxcTz0h8aP4OgLTFqvw?sub_confirmation=1
    promise I'm not just trying to promote my stuff

**Comment by rokm** [comment]
You're not giving us much to work with here...

> I got the PyInstaller to work perfectly with only the ".exe" file in an alt location on my computer, but then I tried to transfer it via remote access to my desktop, and it didn't work.

What was the error message?

> add more data to the ".exe" file to run standalone on an alt computer (no python installed) or from a USB. or...

What data would that be?

> add a "virtual python" or "virtual anaconda" to be placed next to the .exe (or somewhere else) that the .exe can use to run
properly

Why not just install python or anaconda (or transfer a virtual environment) to the alt computer, then?

---

It sounds like you ended up with an incomplete frozen bundle that's missing one or more files that are available on your build system but not on the other one. There are various reasons why that might happen, but without seeing the error message or knowing what your program does (what packages it uses), we cannot really help you there...

**Comment by drakeredwind01** [comment]
first sorry for not getting back sooner and thank you for the quick response (I guess I don't get notified for github stuff).
I am using: 
    time
    pyautogui
    pynput.keyboard
all my script does is look for a specific image then click on it
I am not getting an error message it also doesn't click on the image when it is visible. just a black box. it doesn't matter if I minimize it or move it away from the image. I suspect it is constantly looking for the image but not finding it for some reason. it does work on the computer where I created the script originally.

**Reply by drakeredwind01** [reply]
Please "@" me so I get notified

**Reply by rokm** [reply]
@drakeredwind01 Can you provide a minimal code example that reproduces the issue?

**Reply by drakeredwind01** [reply]
@rokm Thanks for the response
I am not receiving an error or anything of the kind. It simple just sits there (presumably looking for the image).
the code is below:

    if (pyautogui.locateOnScreen('C:\\Users\\drakeredwind01\\Desktop\\sources\\Login.png',confidence=0.8) is not None):
        Login = True
        hit = pyautogui.locateCenterOnScreen(
            'C:\\Users\\drakeredwind01\\Desktop\\sources\\Login.png', confidence=0.8)
        print('hit Login')
        pyautogui.click(hit)

I am also not receiving anything that tells me the image is not where it is supposed to be (which happens if I name the ".png" incorrectly,
for example: 

    OSError: Failed to read C:\Users\drakeredwind01\Desktop\sources\WRONGFILE.png because file is missing, has improper permissions, or is an unsupported or invalid format

so I assume the PyInstaller grabs the image, packs it into the ".exe", and redirects the directory to the new ".png" location.

**Reply by rokm** [reply]
@drakeredwind01 

> so I assume the PyInstaller grabs the image packs it into the ".exe" and redirects the directory to the new ".png" location.

PyInstaller does not automatically collect non-python resources, such as image files (except from packages for which hooks are written). So if `Login.png` is supposed to be part of the bundle, [you need to ensure it is collected](https://pyinstaller.readthedocs.io/en/stable/spec-files.html#adding-files-to-the-bundle). 

Also, in your snippet, you use hard-coded absolute path; that won't work in a frozen application, unless you copy the image to `'C:\\Users\\drakeredwind01\\Desktop\\sources\\Login.png'` on your target computer as well). You should be using [paths anchored to the bundle directory](https://pyinstaller.readthedocs.io/en/stable/runtime-information.html#using-file) to refer to the collected files.

**Reply by drakeredwind01** [reply]
@rokm 
Ok, thank you, I will check out the "Adding Files to the Bundle" and adding "paths anchored to the bundle directory".
I hope at some point there will be a way to add everything into one self-contained ".exe"
